app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2141,4 +2143,32 @@ public void onClick(View view) {
     public enum Flag {
         RESET_DATABASE,
     }
+    public void setupNetworkBroadcastReceiver()
+    {
+        BroadcastReceiver networkStateReceiver = new BroadcastReceiver() {
+            final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.d("Network Listener", "Network Type Changed");
+                boolean isConnected = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
+                if(isConnected){
+                    //Disable fab when there is no connectivity
+                    fab.setEnabled(false);
```

Comment:
Here I was thinking that the chat box would change to fab on scrolling and once it changes to fab we cant go back to chat box as fab is disabled.